Convert submission buttons to fancy radio buttons

emersonford opened the pull request:
We were running into issues where judges accidentally clicked "Previous" or "Current" when they didn't intend to. To remedy this, @MitchTalmadge changed the "Previous", "Current", and "Skip" buttons to radio buttons with a "Submit" button to confirm their choice. 

While this is a pretty small change, we thought it might be a worthwhile quality of life improvement to push upstream. See the attached gif for what the changes look like.

![gavel](https://user-images.githubusercontent.com/1847933/93170533-a5089180-f6e4-11ea-8067-cb386ed3afa6.gif)

anishathalye commented on gavel/templates/vote.html:
```diff
@@ -28,11 +28,31 @@ <h1>Current</h1>
+
+    <script type="application/javascript">
+      function buttonSelected() {
+          $("#vote-done").prop("disabled", false);
```
I'd rather not pull in a large dependency (JQuery) if it's not necessary. Can this be replaced with something like `document.getElementById('vote-done').disabled = false`?

anishathalye replied:
Also, this is the only use of JavaScript in this project, I somehow managed to avoid it otherwise. It would be kinda nice if we could somehow get this functionality without JS, but I don't know if that's easily possible.

Would it be good to have some sort of `<noscript>` somewhere that warns the user if they have JavaScript disabled?

anishathalye commented on gavel/templates/layout.html:
```diff
@@ -7,6 +7,9 @@
     {% endassets %}
 
+    <!-- JQuery -->
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
```
I'd rather avoid dependencies where possible.

anishathalye reviewed (changes requested):
Thank you for the pull request! I appreciate the demo GIF :)

This looks like a great change in terms of usability. Happy to merge it after we're done with the review process. I left some comments on the code.

Do you think it's confusing that "Skip" is a button for the first project, but a radio button for the subsequent ones? Is there a way the two could be unified (or somehow disambiguated)?

emersonford commented:
Done! I removed the jquery dependency, changed the javascript code to accommodate sessions where javascript is disabled, and changed the begin screen to more closely match what is in the vote screen.

When javascript is disabled, the submit button defaults to `disabled = false` and either "Continue" or "Previous" are selected by default.


Demo with Javascript:
![with_js](https://user-images.githubusercontent.com/1847933/93723390-70b92900-fb5b-11ea-8126-0bf80c01704e.gif)

Demo without Javascript:
![without_js](https://user-images.githubusercontent.com/1847933/93723395-7747a080-fb5b-11ea-8c1a-5cf869040e8a.gif)

anishathalye reviewed (changes requested):
Thanks for the changes! I think one last nit: could you change the default from "previous" to "skip" (seems safer) in the case of no JS?

Also, could you squash into a single commit, and then I can merge?

emersonford commented:
Done!

anishathalye commented:
Fantastic. This is a nice usability improvement. Merged! Thanks again for the contribution.
